pkg/services/authn/authnimpl/sync/oauth_token_sync.go

```diff
@@ -113,15 +125,47 @@ func (s *OAuthTokenSync) SyncOauthTokenHook(ctx context.Context, identity *authn
 
 const maxOAuthTokenCacheTTL = 10 * time.Minute
 
-func getOAuthTokenCacheTTL(t time.Time) time.Duration {
-	if t.IsZero() {
+func getOAuthTokenCacheTTL(accessTokenExpiry, idTokenExpiry time.Time) time.Duration {
+	if accessTokenExpiry.IsZero() {
 		return maxOAuthTokenCacheTTL
 	}
 
-	ttl := time.Until(t)
-	if ttl > maxOAuthTokenCacheTTL {
-		return maxOAuthTokenCacheTTL
+	accessTokenTTL := time.Until(accessTokenExpiry)
+	if accessTokenTTL > maxOAuthTokenCacheTTL {
+		accessTokenTTL = maxOAuthTokenCacheTTL
+	}
+
+	if !idTokenExpiry.IsZero() {
+		idTokenTTL := time.Until(idTokenExpiry)
+		if idTokenTTL > maxOAuthTokenCacheTTL {
+			idTokenTTL = maxOAuthTokenCacheTTL
+		}
+		if idTokenTTL < accessTokenTTL {
+			return idTokenTTL
+		}
+	}
+
+	return accessTokenTTL
```

Comment:
This is a small suggestion to improve readability. This is a non-blocking issue.

```suggestion
	if idTokenExpiry.IsZero() {
		return accessTokenTTL
	}

	idTokenTTL := time.Until(idTokenExpiry)
	if idTokenTTL > maxOAuthTokenCacheTTL {
		idTokenTTL = maxOAuthTokenCacheTTL
	}

	if accessTokenTTL <= idTokenTTL {
		return accessTokenTTL
	}
  return idTokenTTL
```